src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3029,6 +3059,7 @@ private void VisitStatementsWithLocalFunctions(BoundBlock block)
         public override BoundNode? VisitLocalFunctionStatement(BoundLocalFunctionStatement node)
```

Comment:
>VisitLocalFunctionStatement [](http://example.com/codeflow?start=35&length=27)

If we expect that the only consumer of this function is ```VisitStatementsWithLocalFunctions```, then  let's make it a local function and let's throw "unreachable" from the override.

Reply:
Looks like this might be called on its own during speculation - see for example test `LocalFunctions_Speculation`.